applications/fishing-map/src/features/workspace/workspace.selectors.ts

```diff
@@ -18,20 +34,78 @@ export const selectWorkspaceViewport = createSelector([selectWorkspace], (worksp
   return workspace?.viewport
 })
 
-export const selectWorkspaceDataviews = createSelector([selectWorkspace], (workspace) => {
-  if (!workspace) return
-  // TODO resolve them here
-  return workspace.dataviews
-})
+export const selectWorkspaceDataviewsResolved = createSelector(
+  [selectWorkspace, selectDataviews],
+  (workspace, urlDataviews = []) => {
+    if (!workspace) return
+    // TODO move this to use-workspace helper
+    const dataviews = workspace.dataviews.flatMap((dataview) => {
+      const workspaceDataviewsConfig = workspace.dataviewsConfig.filter(
+        (dataviewConfig) => dataviewConfig.dataviewId === dataview.id
+      )
+      if (!workspaceDataviewsConfig.length) return dataview
 
-export const selectVesselsDataviews = createSelector([selectWorkspaceDataviews], (dataviews) => {
-  if (!dataviews) return
+      return workspaceDataviewsConfig.map((workspaceDataviewConfig) => {
+        const config = {
+          ...dataview.config,
+          ...workspaceDataviewConfig.config,
+        }
+        const datasetsConfig = dataview.datasetsConfig?.map((datasetConfig) => {
+          const workspaceDataviewDatasetConfig = workspaceDataviewConfig.datasetsConfig?.find(
+            (wddc) =>
+              wddc.datasetId === datasetConfig.datasetId && wddc.endpoint === datasetConfig.endpoint
+          )
+          if (!workspaceDataviewDatasetConfig) return datasetConfig
 
-  return dataviews.filter((dataview) => dataview.config.type === Generators.Type.Track)
-})
+          return { ...datasetConfig, ...workspaceDataviewDatasetConfig }
+        })
+        const resolvedDataview = {
+          ...dataview,
+          config,
+          datasetsConfig,
+        }
+        return {
+          ...resolvedDataview,
+          uid: getUniqueDataviewId(resolvedDataview),
+        }
+      })
+    })
+    // Now it merges with urlDataviews
+    const urlMergedDataviews = dataviews.map((dataview) => {
+      const urlDataview = urlDataviews.find(
+        (urlDataview) =>
+          urlDataview.uid === dataview.id?.toString() || urlDataview.uid === dataview.uid
+      )
+      if (!urlDataview) return dataview
+      return {
+        ...dataview,
+        ...urlDataview,
+        config: {
+          ...dataview.config,
+          ...urlDataview?.config,
+        },
+      }
+    })
+    // Get dataviews defined uniquely in the url
```

Comment:
Why would that ever happen?

Reply:
This is not used anymore as we have a unique identifier for each dataviewConfig